datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -937,14 +935,14 @@ impl<'n> ExprIdentifierVisitor<'_, 'n> {
     ///   information up from children to parents via `visit_stack` during the first,
     ///   visiting traversal and no need to test the expression's validity beforehand with
     ///   an extra traversal).
-    fn pop_enter_mark(&mut self) -> (usize, bool, Option<Identifier<'n>>, bool) {
+    fn pop_enter_mark(&mut self) -> (usize, Option<Identifier<'n>>, bool) {
```

Comment:
Is the information that used to be captured by the 'subtree' boolean flag now kept as part of  `ExprStats`?

Reply:
Previously, we needed the flag as we did stop recursion at the root node of a conditional subtree in `f_down()` and we had to store that piece of information in `visit_stack` so as to remember to compute the hash of the whole conditional subtree in `f_up()` (when we built the root node's identifier).

We no longer need that flag anywhere as we now recurse into conditional subtrees too.